docs: Extend rpm-ostree relationship a bit (#249)
containers/bootc: docs/relationship.md

## Relationship with rpm-ostree

Today both bootc and rpm-ostree use the [ostree project](https://github.com/ostreedev/ostree-rs-ext)
as a backing model.  Hence, when using a container source,
`rpm-ostree upgrade` and `bootc upgrade` are effectively equivalent;
you can use either command.

However with rpm-ostree (or, perhaps re-framed as
"dnf image"), it will continue to work to e.g. `dnf install`
(i.e. `rpm-ostree install`) on the *client side* system.
In addition there are other client-side mutation commands such as
`rpm-ostree initramfs --enable`.

However, as soon as you mutate the system in this way, `bootc upgrade`

> will error out as it will not understand how to upgrade

Reviewer: does `bootc status` detect this state?

Author: Yes, it's what the `incompatible: true` field represents today.

> However, as soon as you do something like this `bootc upgrade`
> then error out as it will not understand how to upgrade
> the system.  The bootc project currently takes a relatively

Reviewer 2: ```suggestion
However, as soon as you mutate the system like this, `bootc upgrade`
will error out as it will not understand how to upgrade
the system.  The bootc project currently takes a relatively
```

Small rewording for clarity

Reviewer 3: s/will error/errors/g

hard stance that system state should come from a container image.

The way kernel argument work also uses ostree on the backend
in both cases, so using e.g. `rpm-ostree kargs` will also work

> on a system updating via bootc.

Reviewer 2: I'm not fully understanding what is being said here...it seems to conflict with what was previously said about client-side mutation.  We want users to define kargs as part of the container image, correct?

Author: Most of the time, yes.  But there are also valid use cases for machine-specific kernel arguments, so that continues to work.
